CMake/resolve_dependency_modules/folly/CMakeLists.txt

```diff
@@ -36,10 +36,10 @@ FetchContent_Declare(
   PATCH_COMMAND git apply ${CMAKE_CURRENT_LIST_DIR}/folly-no-export.patch
                 ${glog_patch})
 
-if(ON_APPLE_M1)
-  # folly will wrongly assume x86_64 if this is not set
-  set(CMAKE_LIBRARY_ARCHITECTURE aarch64)
+if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64")
+  set(IS_AARCH64_ARCH TRUE)
```

Comment:
@zuyu Are you able to confirm from the build that this change works? Can you share the log?

Reply:
@majetideepak W/ the following changes:  
```
if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64")
   set(IS_AARCH64_ARCH TRUE)
   message(STATUS "set(IS_AARCH64_ARCH TRUE)")
 endif()
```

`make debug` output on macOS:
```CMake
-- Setting folly source to AUTO
CMake Warning at CMake/ResolveDependency.cmake:70 (find_package):
  By not providing "Findfolly.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "folly", but
  CMake did not find one.

  Could not find a package configuration file provided by "folly" with any of
  the following names:

    follyConfig.cmake
    folly-config.cmake

  Add the installation prefix of "folly" to CMAKE_PREFIX_PATH or set
  "folly_DIR" to a directory containing one of the above files.  If "folly"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:502 (velox_resolve_dependency)


-- Building Folly from source
-- set(IS_AARCH64_ARCH TRUE)
...
-- Setting FOLLY_USE_SYMBOLIZER: OFF
-- Setting FOLLY_HAVE_ELF:
-- Setting FOLLY_HAVE_DWARF: FALSE
...
-- Using BUNDLED folly
```

Note that
* `IS_AARCH64_ARCH` is defined in [`v2024.07.15.00`](https://github.com/facebook/folly/commit/c30d49dcdc877b38d99b253b8c66ad1853085e09), so it has no effects for now (`v2024.07.01.00`).
* `CMAKE_LIBRARY_ARCHITECTURE` is only used in `message` cmds in folly CMake. In other words, it has no effects at all.

Reply:
Sorry to come in late, this whole if can be removed because I upstreamed the change to folly last year https://github.com/facebook/folly/blob/main/CMakeLists.txt#L82

Reply:
@assignUser I get your point, but on macOS `${CMAKE_SYSTEM_PROCESSOR}` is `arm64`, NOT `aarch64`.

Reply:
No, I do not have any build / link issues.

Just to remove `ON_APPLE_M1`, which was set in `root/CMake`, but only used in `folly/CMake`.

Btw, https://github.com/facebook/folly/commit/c30d49dcdc877b38d99b253b8c66ad1853085e09 is part of `v2024.07.15.00`, and Velox uses `v2024.07.01.00` for now.

Reply:
Right, so we can just remove the entire thing :)

Reply:
@assignUser Plz also review https://github.com/facebook/folly/pull/2341. Ty!